test/test-parse-expression-statement.cpp

```diff
@@ -1369,6 +1369,46 @@ TEST(test_parse, let_as_statement_body_allows_asi) {
                 ElementsAre(spy_visitor::visited_variable_assignment{u8"x"}));
   }
 }
+
+TEST(test_parse, disallow_await_parameter_in_async_arrow_function) {
+  {
+    spy_visitor v;
+    padded_string code(u8"(async (await) => null)"_sv);
```

Comment:
What if `(async (await) => null)` is found within an async function? Would it report multiple errors?

The following code produces two errors, but I think your commit will make it produce three errors:
```js
async function f() {
  (async (await) => null);
}
```
```
hello.js:2:11: error: missing operand for operator [E026]
hello.js:2:11: error: cannot declare 'await' inside async function [E069]
```

Maybe we should avoid calling `parse_expression` if we see an `await` token?

Man, this stuff is confusing.